### DISCUSSION TITLE
Problem to run system tests on Rails

### DISCUSSION THREAD

**Discussion by Pepan** [community/community#26674] [Actions]
<p>My setup file look slike this:</p>
<pre><code>name: CI
on: [push, pull_request]
jobs:
  test:
    runs-on: ubuntu-latest
    services:
      db:
        image: postgres:11
        ports: ['5432:5432']
        options: &gt;-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5
    steps:
      - uses: actions/checkout@v1
      - name: Setup Ruby
        uses: actions/setup-ruby@v1
        with:
          ruby-version: 2.6.x
      - uses: borales/actions-yarn@v2.0.0
        with:
          cmd: install
      - name: Build
        env:
          RAILS_ENV: test
          RAILS_MASTER_KEY: ${{ secrets.RAILS_MASTER_KEY }}
        run: |
          sudo apt-get -yqq install libpq-dev
          cp config/options.demo.yml config/options.yml
          cp config/database.demo.yml config/database.yml
          gem install bundler
          bundle install --jobs 4 --retry 3
      - name: Setup DB
        env:
          DATABASE_URL: postgres://postgres:@localhost:5432/test
          RAILS_ENV: test
          RAILS_MASTER_KEY: ${{ secrets.RAILS_MASTER_KEY }}
        run: |
          bundle exec rails db:create db:migrate
      - name: Run tests
        env:
          DATABASE_URL: postgres://postgres:@localhost:5432/test
          RAILS_ENV: test
          RAILS_MASTER_KEY: ${{ secrets.RAILS_MASTER_KEY }}
        run: |
          bundle exec rails test
          bundle exec rake test:system
</code></pre>
<p>it fails with message as:</p>
<pre><code>==============ERROR["test_showing_a_Page_element", #&lt;Minitest::Reporters::Suite:0x00005564e192b788 @name="PageElementsTest"&gt;, 3.878157251999994]

 test_showing_a_Page_element#PageElementsTest (3.88s)

Selenium::WebDriver::Error::UnknownError: Selenium::WebDriver::Error::UnknownError: unknown error: Chrome failed to start: exited abnormally

          (unknown error: DevToolsActivePort file doesn't exist)

          (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)

            test/application_system_test_case.rb:7:in `log_as_admin'
            test/system/page_elements_test.rb:6:in `block in &lt;class:PageElementsTest&gt;'

==============|
</code></pre>
<p>Trying to add chromium-chromedriver failed with:</p>
<pre><code>sudo apt-get -yqq install libpq-dev chromium-chromedriver
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/universe/c/chromium-browser/chromium-codecs-ffmpeg-extra_77.0.3865.90-0ubuntu0.18.04.1_amd64.deb 404 Not Found [IP: 52.177.174.250 80]
</code></pre>
<p>But without systemtest are GH Actions unusable for Rails dev. :frowning:</p>
<p>Any suggestins / soluttions ?</p>
<hr>
<p>* Also there is some kind of caching missing to speedup gems installation (it takes several minutes each run) - but it is for another post … :/ </p>

**Comment by BrightRan** [comment, marked as answer]
<p>Hi <a class="mention" href="/pepan">@pepan</a> ,</p>
<p>Please download the latest chrome drive from here: <a href="https://sites.google.com/a/chromium.org/chromedriver/downloads" rel="nofollow noopener">https://sites.google.com/a/chromium.org/chromedriver/downloads</a> .<br>
And then try this:</p>
<pre><code>chrome_options = Options()
chrome_options.add_argument('--no-sandbox')
chrome_options.add_argument('--headless')
chrome_options.add_argument('--disable-dev-shm-usage')
d = webdriver.Chrome('/home/PycharmProjects/chromedriver',chrome_options=chrome_options)
d.get('https://www.google.nl/')
</code></pre>

**Comment by lorman** [comment]
<p>Was this ever solved? I’m getting some Chrome Driver errors and would like to know if you had any success with running your specs.</p>

**Comment by ianrandmckenzie** [comment]
<p>Would you be willing to explain what I do after downloading the latest Chromedriver?</p>
<p>I tried adding it to my steps:</p>
<pre><code>- name: Install Chromedriver
  run: |
    # Install Chromedriver
    CHROME_DRIVER_VERSION=`curl -sS https://chromedriver.storage.googleapis.com/LATEST_RELEASE`
    sudo wget -N https://chromedriver.storage.googleapis.com/$CHROME_DRIVER_VERSION/chromedriver_linux64.zip -P ~/
    unzip ~/chromedriver_linux64.zip -d ~/
    sudo mv -f ~/chromedriver /usr/local/bin/chromedriver
    sudo chown root:root /usr/local/bin/chromedriver
    sudo chmod 0755 /usr/local/bin/chromedriver
</code></pre>
<p>Adding all of this, plus adding your recommended configurations didn’t solve the problem.</p>

**Comment by lorman** [comment]
<p>I tweaked my selenium settings in the ruby helper, but you should be able to do the same in most languages I’d guess. </p>
<pre><code>driven_by :selenium, using: :chrome, options: { args: ["headless", "disable-gpu", "no-sandbox", "disable-dev-shm-usage"] }
</code></pre>

**Comment by jywarren** [comment]
<p>I’ve been using the nanasess “setup-chromedriver” action and noticed i was on v1.0.1, but there is now v1.0.7 so maybe that’s also a way to upgrade: <a href="https://github.com/nanasess/setup-chromedriver" class="inline-onebox" rel="noopener nofollow ugc">GitHub - nanasess/setup-chromedriver: ChromeDriver for use in GitHub Actions</a></p>
<p>UPDATE: no, that didn’t work for me.</p>
